subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DetachedConfigurationsIntegrationTest.groovy

```diff
@@ -43,15 +41,12 @@ class DetachedConfigurationsIntegrationTest extends AbstractIntegrationSpec {
                 repositories {
                     maven { url "${mavenRepo.uri}" }
                 }
-                task checkDependencies {
-                    doLast {
-                        configurations.each { conf ->
-                            def declared = conf.dependencies
-                            def detached = project.configurations.detachedConfiguration(declared as Dependency[])
-                            def resolved = detached.resolvedConfiguration.getFirstLevelModuleDependencies()
-                            assert declared*.name == resolved*.moduleName
-                        }
-                    }
+
+                configurations.each { conf ->
```

Comment:
It seems odd to update this test to perform assertions outside of a task. I get that this works but it seems inconsistent with what we do usually. Granted I am not sure why we don't do this more often, but I'm curious what @big-guy thinks here

Applies elsewhere

Reply:
We do this pretty often I thought, for tests that don't require an execution phase.